cachi2/core/package_managers/pip.py

```diff
@@ -204,20 +201,34 @@ def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, str]:
         name = name or setup_cfg.get_name()
         version = version or setup_cfg.get_version()
 
-    if name:
-        log.info("Resolved package name: %r", name)
+    if not name:
+        log.info("Processing metadata from git repository")
+        try:
+            repo_path = get_repo_id(package_dir.root).parsed_origin_url.path.removesuffix(".git")
+            repo_name = Path(repo_path).name
+            package_subpath = package_dir.subpath_from_root
+
+            resolved_path = Path(repo_name).joinpath(package_subpath)
+            normalized_path = pkg_resources.safe_name(str(resolved_path))
+            name = normalized_path.strip("-.")
+        except (UnsupportedFeature, InvalidGitRepositoryError):
```

Comment:
ad `InvalidGitRepositoryError` - if we want to handle that, it should be handled already in `get_repo_id`